### DISCUSSION TITLE
Error 405 when trying create a shortlink

### DISCUSSION THREAD

**Discussion by nacesprin** [shlinkio/discussions#110] [Help wanted] [answered]
Installed through docker:
```
shlinkio/shlink:stable
shlinkio-web-client/shlink-web-client:latest
```
All config are the default settings, so no database engine used than the default.

Error 405 when trying to create a shortlink:
![image](https://user-images.githubusercontent.com/295375/164762869-acdf3b4a-7a5b-4482-bbd8-f9a1508b6a0f.png)
![image](https://user-images.githubusercontent.com/295375/164762723-bdd6c730-ea99-4c26-a711-aaa507488d3a.png)
![image](https://user-images.githubusercontent.com/295375/164762963-5cc27005-7091-46e1-a4f5-e3911af5329b.png)

**Comment by acelaya** [comment, marked as answer]
405 usually means that you are reaching a valid URL, but with the wrong HTTP method.

Considering it's an nginx error, it looks like you didn't configure the vhost correctly, as the request is failing before reaching Shlink.

**Reply by nacesprin** [reply]
Thanks @acelaya .

This is my nginx config. I am grateful if you can help me with the config in all you can. Very thanks.

```
upstream shlinkio {
        server 127.0.0.1:8082;
}
server {
  server_name   link.nacesprin.com;
  location / {
    proxy_pass http://shlinkio/;
    proxy_http_version 1.1;
    proxy_set_header   Host               $host:$server_port;
    proxy_set_header   X-Real-IP          $remote_addr;
    proxy_set_header   X-Forwarded-For    $proxy_add_x_forwarded_for;
    proxy_set_header   X-Forwarded-Proto  $scheme;
    proxy_set_header   Upgrade            $http_upgrade;
    proxy_set_header   Connection         "upgrade";
  }
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/link.nacesprin.com/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/link.nacesprin.com/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot
}

server {
    if ($host = link.nacesprin.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot
  listen        80;
  server_name   link.nacesprin.com;
    return 404; # managed by Certbot
}
```

**Reply by acelaya** [reply]
Try this configuration. It's the one I use and works like a charm https://shlink.io/documentation/advanced/exposing-through-reverse-proxy/#configure-the-proxy

Also, if you can share more details on how you set-up Shlink, it will be easier to help you.

**Reply by socketblind** [reply]
Check that you have added the right API URL to your docker config file!
I also forgotten to add the proto to the beginning and it pointed to the web client not the API server. Same error occured. If you haven't figured it out since April. :)
SHLINK_SERVER_URL: "**http(s)://**shlink.contoso.co" # obviously with the right http or https proto

**Comment by danielthedifficult** [comment]
Thanks for this! In my case, it was the docker host's nginx configuration that was forwarding traffic for all paths to 8081, which was the web client. Adding a rule to redirect anything sent to /rest/ to :8080 (the actual shlink app) fixed it.
